### PR TITLE
[JENKINS-53179] Extend env variable to facilitate the access to the BO URLs

### PR DESCRIPTION
**Pull request by v1v**
[JENKINS-53179](https://issues.jenkins-ci.org/browse/JENKINS-53179)

Enable URLs for a page that displays tests and artifacts for a Run.

Depends on https://github.com/jenkinsci/display-url-api-plugin/pull/25

## Highlights
- `RUN_ARTIFACTS_DISPLAY_URL` points out to the qualified URL for a page that displays artifacts for a Run
- `RUN_TESTS_DISPLAY_URL ` points out to the qualified URL for a page that displays tests for a Run

## UI

```groovy
node {
    sh 'env | grep -i display_ | sort | tee file.txt'
    archive 'file.txt'
}
```

image

image

image

@jenkinsci/code-reviewers

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -73,7 +73,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>display-url-api</artifactId>
-      <version>2.3.1</version>
+      <version>2.3.2-SNAPSHOT</version>
```
Comment (v1v) [inline pom.xml]: TODO: Change the version with the incremental one

Comment (v1v) [reply pom.xml]: As long as https://github.com/jenkinsci/display-url-api-plugin/pull/26 is merged then https://github.com/jenkinsci/display-url-api-plugin/pull/25 will generated an incremental artifact which could be used in this case

File: src/test/java/org/jenkinsci/plugins/blueoceandisplayurl/BlueOceanDisplayURLImplTest.java [hunk 2]
```diff
@@ -223,7 +251,6 @@ public static MultiBranchTestBuilder createProject(JenkinsRule j, String name, G
             return new MultiBranchTestBuilder(j, mp);
         }
 
-
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/blueoceandisplayurl/BlueOceanDisplayURLImplTest.java]: Please avoid making formatting changes to unrelated parts of a file. PR diffs should be as short as possible to accomplish the stated goal.

**Comment by abierbaum** [comment]
@v1v @jglick Was just looking for something exactly like this today.  Any chance to get through through and into the plugin?

**Comment by StevenGiguere76** [comment]
Anything this patch needs before it can be merged in? Lots of articles pointing to the fact it needs to be solved, has been, but hasn't been pushed fully into the plugin?

**Comment by v1v** [comment]
Need to address some of the concerns in https://github.com/jenkinsci/display-url-api-plugin/pull/25 in order to move forward

File: pom.xml [hunk 3]
```diff
@@ -72,7 +72,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>display-url-api</artifactId>
-      <version>2.3.2</version>
+      <version>2.3.3-rc125.86e54dc35a15</version>
```
Comment (v1v) [inline pom.xml]:
```suggestion
      <version>2.3.3-rc129.d96f21b00c17</version>
```

File: pom.xml [hunk 4]
```diff
@@ -72,7 +72,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>display-url-api</artifactId>
-      <version>2.3.2</version>
+      <version>2.3.3-rc129.d96f21b00c17</version>
```
Comment (v1v) [inline pom.xml]:
```suggestion
      <version>2.3.3-rc129.d96f21b00c17</version>
```

**Comment by FallDownT** [comment]
Is anything still needed for this patch to be merged in?  This feature would be extremely useful to me right now and it seems ready to merge.

**Comment by v1v** [comment]
I'm just a contributor,  the maintainer is the one to review this PR and merge if happy with

**Comment by v1v** [comment]
@jenkinsci/code-reviewers  this PR has been stalled for a year or so, any feedback?

**Comment by olamy** [comment]
@v1v can you please rebase? I would prefer a released version of https://github.com/jenkinsci/display-url-api-plugin but that's my job to do it I guess ;)

**Comment by jglick** [comment]
> I would prefer a released version of https://github.com/jenkinsci/display-url-api-plugin but that's my job to do it I guess

Certainly this should not be merged while referring to an incremental dependency version. What I have done in the past as a plugin maintainer, or recommended other maintainers do: release the upstream plugin, then use GH suggest changes to guide the downstream PR author to switch to the new release version. At that point downstream is safe to merge & release. (No need for the PR author to rebase; maintainer can always squash-merge if they enjoy destructive history operations.)

**Comment by olamy** [comment]
@v1v version 2.3.3 done please update your pull request thanks

File: pom.xml (hunk 4 above)
Comment (jglick) [inline pom.xml:75]: thus
```suggestion
      <version>2.3.3</version>
```

**Comment by jglick** [comment]
(plus merge conflict resolution)

**Comment by timja** [comment]
@olamy or @jglick could this be released please?

**Comment by olamy** [comment]
@timja oh yes I will do it early next week

**Comment by syl20bnr** [comment]
@olamy When do you have time to release the version 2.4 of the plugin ?

**Comment by olamy** [comment]
@syl20bnr  all done buddy

**Comment by syl20bnr** [comment]
Thank you ! 💯

**Pull request merged**